docker_cli/commands.py

```diff
@@ -0,0 +1,119 @@
+    """
+    Initiates a drop-in session on an image.
+
+    Parameters
+    ----------
+    tag : str
+        The tag or ID of the image.
+    """
+    image: Image = Image(tag)
+
+    image.drop_in()
+
+
+def remove(
+    tags: Iterable[str],
+    force: bool = True,
+    verbose: bool = False,
+    ignore_prefix: bool = False,
+) -> None:
+    """
+    Remove all Docker images that match a given tag or wildcard pattern.
+
+    This tag or wildcard will take the form [UNIVERSAL PREFIX]-[tag or wildcard] if the
+    prefix does not already match this.
+
+    Parameters
+    ----------
+    tags : Iterable[str]
+        An iterable of tags or wildcards to be removed.
+    force : bool, optional
+        Whether or not to force the removal. Defaults to True.
+    verbose : bool, optional
+        Whether or not to print output for removals verbosely. Defaults to False.
+    ignore_prefix: bool, optional
+        Whether or not to ignore the universal prefix and only use the tag or wildcard.
+        Use with caution, as this will remove ALL images matching the wildcard.
+        e.g. ``remove(["*"], ignore_prefix = True)`` will remove all images.
+    """
+    force_arg = "--force " if force else ""
+
+    # The None below corresponds to printing outputs to the console. DEVNULL causes the
+    # outputs to be discarded.
+    output = None if verbose else DEVNULL
+
+    # Search for and delete all images matching each tag or wildcard.
+    for tag in tags:
+        prefix = universal_tag_prefix()
+        search = tag if (tag.startswith(prefix) or ignore_prefix) else f"{prefix}-{tag}"
+        if verbose:
+            print(f"Attempting removal for tag: {search}")
+
+        # Search for all images whose name matches this tag, acquire a list
+        search_command = split(f'docker images --filter=reference="{search}" -q')
+        search_result = run(search_command, text=True, stdout=PIPE, stderr=output)
+        # An empty return indicates that no such images were found. Skip to the next.
+        if search_result.stdout == "":
+            if verbose:
+                print(f"No images found matching pattern {search}. Proceeding.")
+            continue
+        # The names come in a list delimited by newlines. Reform this to be delimited
+        # by spaces to use with `Docker rmi`.
+        search_result_str = search_result.stdout.replace("\n", " ")
+
+        # Remove all images in the list
+        command = split(f"docker rmi {force_arg}{search_result_str}")
+        run(command, stdout=output, stderr=output)
+    print("Docker removal process completed.")
+
+
+def make_lockfile(
+    tag: str, file: Union[str, os.PathLike[str]], env_name: str = "base"
+) -> None:
+    """
+    Makes a lockfile from an image.
+
+    ..warning:
+        This function only works for images that have an environment set up.
+
+    Parameters
+    ----------
+    tag : str
+        The tag or ID of the image.
+    filename : Union[str, os.PathLike[str]]
```

Comment:
This parameter is currently named "file"

```suggestion
    file : Union[str, os.PathLike[str]]
```